<h1 style="color:#ffffff; background-color:#2E6B4A; text-align:center; font-weight:bold; padding:16px 10px; border-radius:8px; margin-bottom:6px;">Week 6 — Day 1: Sprint 1 Planning &amp; Neural Network Architecture</h1>
<p style="color:#F2C14E; background-color:#2E6B4A; text-align:center; font-weight:bold; font-style:italic; padding:6px 10px; border-radius:6px; margin-top:0; font-size:15px;">Phase 3, Sprint 1 — Cardiac Patient Monitoring System</p>

<a id="toc"></a>
<a id="toc"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">Table of Contents</h2>
<div style="border:1px solid #ccc; padding:15px 25px; border-radius:6px;">
<ol style="font-weight:bold; line-height:1.9;">
<li><a href="#section0">0. Setup</a></li><li><a href="#section1">1. Sprint 1 Planning &amp; the Baseline First</a></li><li><a href="#section2">2. Connecting Week 5 to Week 6 — Where This Project Stands</a></li><li><a href="#section3">3. Why Deep Learning</a></li><li><a href="#section4">4. The Neuron</a></li><li><a href="#section5">5. Layers and Architecture</a></li><li><a href="#section6">6. Weights and Biases Are What's Learned</a></li><li><a href="#section7">7. Common Mistakes to Avoid</a></li><li><a href="#section8">8. Quick Reference — Cheat Sheet</a></li><li><a href="#section9">9. Hands-On Lab — Sprint 1 Kickoff &amp; Baseline</a></li><li><a href="#section10">10. Best Practices &amp; Reproducibility</a></li><li><a href="#section11">11. Summary — What I Learned Today</a></li>
</ol>
</div>

<a id="section0"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">0. Setup</h2>
<div style="border-left:5px solid #4A9DE0; background-color:#eaf4fc; padding:10px 15px; margin:10px 0; border-radius:4px; color:#1c4e6e;">
<b>Note:</b> This notebook opens Sprint 1 of Phase 3 for the <b>Cardiac Patient Monitoring System</b> capstone project (selected and planned in Week 5, Day 5). Rather than starting a new dataset from scratch, Day 1 confirms the Sprint 1 plan and formally records the baseline that already exists from the project's classical machine learning phase — the exact baseline this week's neural network must beat.
</div>

In [1]:
import pandas as pd
import numpy as np

print('Pandas version:', pd.__version__)
print('NumPy version :', np.__version__)

Pandas version: 2.3.3
NumPy version : 2.3.5


<a id="section1"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">1. Sprint 1 Planning &amp; the Baseline First</h2>
Day 1 opens with formal <b>Sprint Planning</b>: confirm the Sprint 1 goal set in Week 5, break it into backlog tasks, and commit to the goal. Sprint 1's real objective is to establish a <b>baseline</b> — a simple model (from Weeks 3–4) whose score every later model, including this week's neural network, must beat. Without a baseline, a neural network's score is meaningless.

<div style="border-left:5px solid #9B6FD4; background-color:#f3edfb; padding:10px 15px; margin:10px 0; border-radius:4px; color:#4d3475;">
<b>Goal:</b> <b>Sprint 1 goal (confirmed from Week 5):</b> "Understand the churn dataset" was the generic template — for this project specifically, the goal is restated as: <i>"Confirm the existing classical-ML baseline for heart-disease classification, and establish it as the score the Sprint 1 neural network must beat."</i>
</div>

<a id="section2"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">2. Connecting Week 5 to Week 6 — Where This Project Stands</h2>
It is worth being explicit about the relationship between last week's planning day and this week's first execution day, since it is easy to lose track of it once several weeks have passed:

<div style="border:1px solid #ccc; border-radius:6px; overflow:hidden;">
<table style="width:100%; border-collapse:collapse;">
<tr style="background-color:#2E6B4A; color:#F2C14E;"><th style="padding:8px; text-align:left;"></th><th style="padding:8px; text-align:left;">Week 5, Day 5</th><th style="padding:8px; text-align:left;">Week 6, Day 1</th></tr>
<tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">What happened</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Planning only — no model code written</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">First real execution of the Sprint 1 backlog</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Deliverable</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Project selection, problem statement, Sprint 1 backlog</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Confirmed goal + a recorded, trusted baseline score</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Model status</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">No model trained yet</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Baseline model trained and evaluated</td></tr>
</table>
</div>

For this specific project, the Sprint 1 backlog item <i>"train a simple baseline model"</i> was already completed ahead of schedule during the project's own classical machine learning phase — the notebook <code>02_supervised_learning_and_model_comparison.ipynb</code> already trained and evaluated five classical models, including a Logistic Regression baseline. Day 1's job today is not to retrain that baseline from scratch, but to <b>formally confirm it as the Sprint 1 baseline</b> and record its exact score before any neural network work begins later this week.

<div style="border-left:5px solid #E85D9A; background-color:#fdeef4; padding:10px 15px; margin:10px 0; border-radius:4px; color:#8a2b52;">
<b>Tip:</b> Being ahead of schedule is not a problem — it just means Day 1's "train a baseline" backlog task becomes "confirm and formally record the existing baseline." The Sprint discipline (goal, backlog, recorded score, feature branch, PR) still applies exactly the same way.
</div>

<a id="section3"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">3. Why Deep Learning</h2>
Classical ML (Weeks 3–4), which this project's five-model comparison already used, works well on structured, tabular data — exactly the kind of data in <code>heart.csv</code>. Deep learning shines where classical methods struggle: images, text, audio, and other high-dimensional, unstructured data with complex patterns. A neural network learns its own features directly from raw data, which is why it powers modern computer vision and NLP — the topics of Weeks 7–8.

<div style="border-left:5px solid #E85D9A; background-color:#fdeef4; padding:10px 15px; margin:10px 0; border-radius:4px; color:#8a2b52;">
<b>Tip:</b> This project's own model comparison already suggests a neural network may not have much room to improve: five very different classical algorithms (Logistic Regression, Decision Tree, Random Forest, SVM, KNN) all landed in a fairly tight accuracy band (82%–89%). That said, the comparison is only valid if it is actually run — which is exactly what the rest of this Sprint does.
</div>

<a id="section4"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">4. The Neuron</h2>
A single neuron is the building block, and it is exactly the operation from Week 2: it takes inputs, multiplies each by a weight, sums them, adds a bias, then passes the result through an activation function. In one line: <code>output = activation( dot(inputs, weights) + bias )</code>. A neural network is just many of these neurons connected in layers.

In [2]:
# One neuron, conceptually, using two of this project's own scaled features:
# z = (Oldpeak * w1) + (MaxHR * w2) + bias      <- the dot product from Week 2
# output = activation(z)

import numpy as np

# Example values, standardized the same way the project's preprocessing pipeline does
oldpeak_scaled = 0.8   # a patient with an above-average Oldpeak
maxhr_scaled = -1.1    # a patient with a below-average max heart rate

x = np.array([oldpeak_scaled, maxhr_scaled])
w = np.array([0.6, -0.5])   # made-up starting weights
bias = 0.05

z = np.dot(x, w) + bias
output = 1 / (1 + np.exp(-z))   # sigmoid activation -> a probability

print('Weighted sum + bias (z):', round(z, 4))
print('Neuron output (probability of HeartDisease):', round(output, 4))

Weighted sum + bias (z): 1.08
Neuron output (probability of HeartDisease): 0.7465


<div style="border-left:5px solid #4A9DE0; background-color:#eaf4fc; padding:10px 15px; margin:10px 0; border-radius:4px; color:#1c4e6e;">
<b>Note:</b> Notice both features chosen here — <code>Oldpeak</code> and <code>MaxHR</code> — are exactly the two numeric features this project's own EDA (from the classical ML phase) found to have the strongest correlation with <code>HeartDisease</code> (+0.404 and -0.401). A real neuron does not know this in advance; it starts with random weights and learns which features matter through training — but it is a useful sanity check that the raw math above already leans the right direction.
</div>

<a id="section5"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">5. Layers and Architecture</h2>
Neurons are organized into layers, and data flows through them in order:

<div style="border:1px solid #ccc; border-radius:6px; overflow:hidden;">
<table style="width:100%; border-collapse:collapse;">
<tr style="background-color:#2E6B4A; color:#F2C14E;"><th style="padding:8px; text-align:left;">Layer</th><th style="padding:8px; text-align:left;">Role</th><th style="padding:8px; text-align:left;">For This Project</th></tr>
<tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Input layer</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Receives the raw features (one node per feature)</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">15 nodes — the same 15 numeric + one-hot encoded columns the classical models already use</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Hidden layer(s)</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Learn intermediate representations; more/wider layers = more capacity</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">To be designed on Day 4</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Output layer</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Produces the final prediction (shape depends on the task)</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">1 node, sigmoid activation — binary HeartDisease probability</td></tr>
</table>
</div>

"Deep" learning simply means a network with multiple hidden layers. Each layer transforms its input and passes the result forward, and stacking layers lets the network learn increasingly abstract patterns.

<a id="section6"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">6. Weights and Biases Are What's Learned</h2>
The weights and biases are the network's parameters — the numbers adjusted during training. A fresh network starts with random weights and makes random predictions; training is the process of nudging these weights until the predictions become accurate. Everything on Days 2–3 this week is about how that nudging works.

<div style="border-left:5px solid #4A9DE0; background-color:#eaf4fc; padding:10px 15px; margin:10px 0; border-radius:4px; color:#1c4e6e;">
<b>Note:</b> The neuron is not new math — it is the Week 2 dot product plus a bias plus an activation. A neural network is that same operation, stacked and repeated. Keeping this in mind demystifies deep learning.
</div>

<a id="section7"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">7. Common Mistakes to Avoid</h2>
<div style="border-left:5px solid #F0964B; background-color:#fef2e7; padding:10px 15px; margin:10px 0; border-radius:4px; color:#8a4a13;">
<b>Important:</b> <ul><li><b>Retraining the baseline from scratch instead of reusing it</b> — if a trustworthy baseline already exists (as it does here), formally confirming and recording it is enough; there is no need to duplicate work.</li><li><b>Assuming a neural network will automatically beat classical models</b> — on small, tabular datasets like this one (917 rows), classical models are often very competitive, sometimes better.</li><li><b>Forgetting the preprocessing pipeline already built</b> — the same <code>ColumnTransformer</code> logic in <code>src/preprocessing.py</code> should feed the neural network too, for a fair comparison.</li><li><b>Skipping formal Sprint Planning because 'the project is basically done'</b> — the Sprint discipline (goal, backlog, feature branches, PRs) still matters for documenting the deep-learning phase properly.</li><li><b>Comparing the neural network to the wrong baseline metric</b> — F1-score and ROC-AUC, not accuracy alone, are this project's agreed primary metrics because the classes are only mildly imbalanced (44.71% / 55.29%).</li></ul>
</div>

<a id="section8"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">8. Quick Reference — Cheat Sheet</h2>

<div style="border:1px solid #ccc; border-radius:6px; overflow:hidden;">
<table style="width:100%; border-collapse:collapse;">
<tr style="background-color:#2E6B4A; color:#F2C14E;"><th style="padding:8px; text-align:left;">Concept</th><th style="padding:8px; text-align:left;">One-Line Reminder</th></tr>
<tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Sprint 1 real goal</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Confirm/establish a baseline; every later model must beat it</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">This project's baseline</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Logistic Regression — 87.50% accuracy, 93.90% ROC-AUC</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">This project's best classical model</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Random Forest — 89.13% accuracy, 92.91% ROC-AUC</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Neuron</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">output = activation(dot(inputs, weights) + bias)</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Input layer size (this project)</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">15 nodes (numeric + one-hot encoded categorical features)</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Output layer (this project)</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">1 node, sigmoid — binary classification probability</td></tr><tr style="background-color:#ffffff;"><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">Deep learning</td><td style="padding:8px; border-top:1px solid #eee; color:#1a1a1a;">A network with multiple hidden layers</td></tr>
</table>
</div>

<a id="section9"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">9. Hands-On Lab — Sprint 1 Kickoff &amp; Baseline</h2>
<div style="border-left:5px solid #9B6FD4; background-color:#f3edfb; padding:10px 15px; margin:10px 0; border-radius:4px; color:#4d3475;">
<b>Goal:</b> Run the five real steps a Sprint 1 kickoff requires, applied to the Cardiac Patient Monitoring project specifically: confirm the plan, load the already-cleaned project dataset, formally record the existing baseline, document the Git workflow, and write down the exact score this week's neural network must beat.
</div>

<h3 style="color:#F2C14E; font-weight:bold; background-color:#2D3748; display:inline-block; padding:4px 10px; border-radius:6px;">Step 1 — Confirm Sprint 1 Planning</h3>

<div style="border-left:5px solid #9B6FD4; background-color:#f3edfb; padding:10px 15px; margin:10px 0; border-radius:4px; color:#4d3475;">
<b>Goal:</b> <b>Sprint 1 goal:</b> "Confirm the existing Logistic Regression baseline on the Cardiac Patient Monitoring dataset, and formally record its score as the target this week's neural network must beat."<br><b>Backlog for today:</b> load the cleaned project dataset → confirm baseline metrics from the existing model-comparison output → document the Git workflow → record the baseline score.
</div>

<h3 style="color:#F2C14E; font-weight:bold; background-color:#2D3748; display:inline-block; padding:4px 10px; border-radius:6px;">Step 2 — Load the Project's Cleaned Dataset</h3>

In [3]:
df = pd.read_csv("heart_cleaned.csv")

print('Shape:', df.shape)
df.head()

Shape: (917, 12)


,Age,Sex,ChestPainType,RestingBP,Cholesterol,FastingBS,RestingECG,MaxHR,ExerciseAngina,Oldpeak,ST_Slope,HeartDisease
0,40,M,ATA,140,289.0,0,Normal,172,N,0.0,Up,0
1,49,F,NAP,160,180.0,0,Normal,156,N,1.0,Flat,1
2,37,M,ATA,130,283.0,0,ST,98,N,0.0,Up,0
3,48,F,ASY,138,214.0,0,Normal,108,Y,1.5,Flat,1
4,54,M,NAP,150,195.0,0,Normal,122,N,0.0,Up,0


In [4]:
print("Target balance (HeartDisease):")
print(df["HeartDisease"].value_counts(normalize=True).round(4))

Target balance (HeartDisease):
HeartDisease
1    0.5529
0    0.4471
Name: proportion, dtype: float64


<div style="border-left:5px solid #4A9DE0; background-color:#eaf4fc; padding:10px 15px; margin:10px 0; border-radius:4px; color:#1c4e6e;">
<b>Note:</b> This is the same cleaned dataset the project's classical ML notebook already used — 917 patients, with the invalid <code>RestingBP = 0</code> row removed and <code>Cholesterol = 0</code> values already flagged for the pipeline's median imputation. No new cleaning is needed here.
</div>

<h3 style="color:#F2C14E; font-weight:bold; background-color:#2D3748; display:inline-block; padding:4px 10px; border-radius:6px;">Step 3 — Confirm the Existing Baseline Score</h3>

In [5]:
model_comparison = pd.read_csv("model_comparison.csv")
model_comparison

,Model,Accuracy,Precision,Recall,F1 Score,ROC-AUC,Mean CV F1,CV F1 Std
0,Random Forest,0.8913,0.9020,0.9020,0.9020,0.9291,0.8732,0.0261
1,SVM (RBF Kernel),0.8804,0.8704,0.9216,0.8952,0.9433,0.8773,0.0296
2,K-Nearest Neighbors,0.8750,0.8762,0.9020,0.8889,0.9295,0.8701,0.0289
3,Logistic Regression,0.8750,0.8835,0.8922,0.8878,0.9390,0.8688,0.0319
4,Decision Tree,0.8207,0.8485,0.8235,0.8358,0.8974,0.8300,0.0197


In [6]:
baseline_row = model_comparison[model_comparison["Model"] == "Logistic Regression"].iloc[0]

print("="*60)
print("CONFIRMED SPRINT 1 BASELINE")
print("="*60)
print(f"Model      : {baseline_row['Model']}")
print(f"Accuracy   : {baseline_row['Accuracy']:.4f}")
print(f"Precision  : {baseline_row['Precision']:.4f}")
print(f"Recall     : {baseline_row['Recall']:.4f}")
print(f"F1 Score   : {baseline_row['F1 Score']:.4f}")
print(f"ROC-AUC    : {baseline_row['ROC-AUC']:.4f}")
print("="*60)

CONFIRMED SPRINT 1 BASELINE
Model      : Logistic Regression
Accuracy   : 0.8750
Precision  : 0.8835
Recall     : 0.8922
F1 Score   : 0.8878
ROC-AUC    : 0.9390


<div style="border-left:5px solid #E85D9A; background-color:#fdeef4; padding:10px 15px; margin:10px 0; border-radius:4px; color:#8a2b52;">
<b>Tip:</b> <b>Why Logistic Regression, and not Random Forest (the project's best classical model)?</b> The baseline is meant to be the <i>simplest</i> reasonable model, not the strongest one. Logistic Regression is the standard Week 3 baseline choice. Random Forest is kept visible as the classical benchmark to also beat, but Logistic Regression is the formal "baseline" for Sprint 1 comparison purposes.
</div>

<h3 style="color:#F2C14E; font-weight:bold; background-color:#2D3748; display:inline-block; padding:4px 10px; border-radius:6px;">Step 4 — Document the Git Workflow</h3>

In [7]:
git_workflow = """
Feature branch : feature/week6-day1-sprint1-baseline
Commit message : "Confirm Sprint 1 baseline for Cardiac Patient Monitoring (Logistic Regression)"
Pull request   : Draft PR opened against main, tagged for mentor review on Day 3
Notebook state : Runs top-to-bottom without errors; baseline metrics printed and documented below
"""
print(git_workflow)


Feature branch : feature/week6-day1-sprint1-baseline
Commit message : "Confirm Sprint 1 baseline for Cardiac Patient Monitoring (Logistic Regression)"
Pull request   : Draft PR opened against main, tagged for mentor review on Day 3
Notebook state : Runs top-to-bottom without errors; baseline metrics printed and documented below



<div style="border-left:5px solid #E85D9A; background-color:#fdeef4; padding:10px 15px; margin:10px 0; border-radius:4px; color:#8a2b52;">
<b>Tip:</b> In the real repository this is an actual <code>git checkout -b feature/week6-day1-sprint1-baseline</code>, <code>git add</code>, <code>git commit</code>, and <code>git push</code>, followed by opening a draft pull request on GitHub — this cell only documents what that workflow looked like, for reproducibility.
</div>

<h3 style="color:#F2C14E; font-weight:bold; background-color:#2D3748; display:inline-block; padding:4px 10px; border-radius:6px;">Step 5 — Record the Baseline Score to Beat</h3>

In [8]:
print("="*60)
print("SPRINT 1 BASELINE — RECORD FOR COMPARISON")
print("="*60)
print("Project    : Cardiac Patient Monitoring System")
print("Dataset    : heart_cleaned.csv (n=917)")
print(f"Baseline   : Logistic Regression")
print(f"Accuracy   : {baseline_row['Accuracy']:.4f}")
print(f"F1 Score   : {baseline_row['F1 Score']:.4f}")
print(f"ROC-AUC    : {baseline_row['ROC-AUC']:.4f}")
print("-"*60)
print("Stretch benchmark (best classical model, Random Forest):")
rf_row = model_comparison[model_comparison["Model"] == "Random Forest"].iloc[0]
print(f"Accuracy   : {rf_row['Accuracy']:.4f}")
print(f"F1 Score   : {rf_row['F1 Score']:.4f}")
print(f"ROC-AUC    : {rf_row['ROC-AUC']:.4f}")
print("="*60)

SPRINT 1 BASELINE — RECORD FOR COMPARISON
Project    : Cardiac Patient Monitoring System
Dataset    : heart_cleaned.csv (n=917)
Baseline   : Logistic Regression
Accuracy   : 0.8750
F1 Score   : 0.8878
ROC-AUC    : 0.9390
------------------------------------------------------------
Stretch benchmark (best classical model, Random Forest):
Accuracy   : 0.8913
F1 Score   : 0.9020
ROC-AUC    : 0.9291


<div style="border-left:5px solid #9B6FD4; background-color:#f3edfb; padding:10px 15px; margin:10px 0; border-radius:4px; color:#4d3475;">
<b>Goal:</b> <b>Baseline to beat: Logistic Regression at 87.50% accuracy and 0.9390 ROC-AUC.</b> <b>Stretch benchmark: Random Forest at 89.13% accuracy and 0.9291 ROC-AUC.</b> Every model built for the rest of this sprint — including Day 4's neural network — is judged against these two numbers, not against "does it seem to work."
</div>

<a id="section10"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">10. Best Practices &amp; Reproducibility</h2>
<div style="border-left:5px solid #F0964B; background-color:#fef2e7; padding:10px 15px; margin:10px 0; border-radius:4px; color:#8a4a13;">
<b>Important:</b> <ul><li>Reuse an already-trustworthy baseline rather than retraining it from scratch — confirming and recording it is enough.</li><li>Keep both the simple baseline (Logistic Regression) and the strongest classical model (Random Forest) visible, since a neural network should ideally be compared against both.</li><li>Reuse the exact same cleaned dataset and preprocessing logic (<code>src/preprocessing.py</code>) for the neural network, so any difference in score reflects the model, not the data.</li><li>Still follow full Sprint discipline (goal, backlog, feature branch, PR) even when a step feels 'already done' — documentation is part of the deliverable.</li><li>Keep the neuron mental model (dot product + bias + activation) in mind going into Days 2–3 — it prevents deep learning from feeling like new math.</li></ul>
</div>

<a id="section11"></a>
<h2 style="color:#F2C14E; background-color:#2E6B4A; font-weight:bold; margin-top:30px; padding:8px 14px; border-radius:6px;">11. Summary — What I Learned Today</h2>
<div style="border-left:5px solid #E85D9A; background-color:#fdeef4; padding:10px 15px; margin:10px 0; border-radius:4px; color:#8a2b52;">
<b>Tip:</b> <ul><li><b>Week 5, Day 5 planned; Week 6, Day 1 executes</b> — today confirmed the Sprint 1 goal and formally recorded the baseline for the Cardiac Patient Monitoring project.</li><li>The confirmed <b>baseline is Logistic Regression at 87.50% accuracy and 0.9390 ROC-AUC</b>, with Random Forest (89.13% accuracy) as a stretch benchmark.</li><li>A <b>neuron</b> is exactly the Week 2 dot product, plus a bias, passed through an <b>activation function</b> — demonstrated here using this project's own <code>Oldpeak</code> and <code>MaxHR</code> features.</li><li>This project's neural network will need a <b>15-node input layer</b> (matching the existing preprocessing) and a <b>1-node sigmoid output layer</b> for binary classification.</li><li><b>Weights and biases</b> are the parameters a network learns — Days 2–3 cover exactly how that learning happens, before the network is actually built on Day 4.</li></ul>
</div>